service/raft/raft_address_map.hh

```diff
@@ -79,20 +77,17 @@ class raft_address_map {
         explicit expiring_entry_ptr(list_type& l, const raft::server_id& entry_id)
             : _expiring_list(l), _last_accessed(Clock::now()), _entry_id(entry_id)
         {
+            _expiring_list.push_front(*this);
         }
 
         ~expiring_entry_ptr() {
-            if (lru_list_hook::is_linked()) {
-                _expiring_list.erase(_expiring_list.iterator_to(*this));
-            }
+            _expiring_list.erase(_expiring_list.iterator_to(*this));
         }
 
         // Update last access timestamp and move ourselves to the front of LRU list.
         void touch() {
             _last_accessed = Clock::now();
-            if (lru_list_hook::is_linked()) {
-                _expiring_list.erase(_expiring_list.iterator_to(*this));
-            }
+            _expiring_list.erase(_expiring_list.iterator_to(*this));
```

Comment:
Well, the invariant is broken between this line and the next one. I wonder what happens of `push` below fails?

Reply:
https://www.boost.org/doc/libs/1_80_0/doc/html/boost/intrusive/list.html#idm35197-bb

```
Requires: value must be an lvalue.

Effects: Inserts the value in the front of the list. No copy constructors are called.

Throws: Nothing.

Complexity: Constant.

Note: Does not affect the validity of iterators and references. 
```

Reply:
Ah, this one is intrusive. OK.